codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -253,6 +253,17 @@
     }
 
     /**
+     * A key to be used for associating application-defined properties with streams within this connection.
+     */
+    interface PropertyKey {
+    }
```

Comment:
Can you explain the rational as to why there is no `int index();` in here?  The cast we are doing for the `addProperty` (and other methods) seems dangerous when we can just have the interface include the methods we depend upon.  Is this cast the mechanism to enforce that the key was registered with `PropertyKeyRegistry`?  If so I think that may be a bad assumption because you can always use keys with different connections and still have issues.